internals/src/num/mod.rs

```diff
@@ -0,0 +1,23 @@
+//! Helpers for integers
+
+/// Extensions for signed integers
+pub trait IntExt {
+    /// the unsigned integer type
+    type Unsigned;
+
+    ///use this to find unsigned absolute if rust version less that 1.51
+    #[cfg(not(rust_v_1_51))]
+    fn unsigned_abs(self) -> Self::Unsigned;
+}
+
+impl IntExt for i64 {
+    type Unsigned = u64;
+
+    fn unsigned_abs(self) -> Self::Unsigned {
+        if self < 0 {
+            -self as u64
+        } else {
+            self as u64
+        }
```

Comment:
I'm not sure if this is better than what's in `std` but I guess the optimizer will see it anyway.

Reply:
Ah bloody hell, just spent five minutes reading all the `abs`, `wrapping_abs`, `unsigned_abs` in the stdlib - I guess doing integer types correctly is hard. 

I think we should do what stdlib does and use 
```rust
pub fn unsigned_abs(x: i8) -> u8 {
    x.wrapping_abs() as u8
}
```
Call chain for input -128 : `wrapping_abs` -> `is_negative` -> `wrapping_neg` -> `wrapping_sub`

All available in our MSRV

phew!

Sorry for not investigating fully when this work started.

Reply:
I've gotta say, this is possibly the worst first issue @jim4067 could've taken up -- sorry!! It has turned out to be 10x as complicated as any of us first expected.

Reply:
@apoelstra its okay. I consider this a great learning experience.